### DISCUSSION TITLE
Fail Sync on Android an Chrome Plugin randomly

### DISCUSSION THREAD

**Discussion by Crosus97** [dani-garcia/vaultwarden#4554] [Q&A]
### Subject of the issue
when press de sync button apper a Fail Sync Error, that work until 3 o 4 days, no changes are made it

### Your environment (Generated via diagnostics page)
* Vaultwarden version: v1.30.5
* Web-vault version: v2024.1.2b
* OS/Arch: linux/x86_64
* Running within a container: true (Base: Debian)
* Environment settings overridden: true
* Uses a reverse proxy: false (**THIS IS TRUE BUT THE DIAGNOSTICS DIDNT DETECT IT, IS HAPROXY FROM PFSENSE**)
* Internet access: true
* Internet access via a proxy: false
* DNS Check: true
* Browser/Server Time Check: true
* Server/NTP Time Check: true
* Domain Configuration Check: true
* HTTPS Check: true
* Database type: SQLite
* Database version: 3.44.0
* Clients used: 
* Reverse proxy and version: 
* Other relevant information: 

### Config (Generated via diagnostics page)
<details><summary>Show Running Config</summary>

**Environment settings which are overridden:** ADMIN_TOKEN

```json
{
  "_duo_akey": null,
  "_enable_duo": true,
  "_enable_email_2fa": false,
  "_enable_smtp": true,
  "_enable_yubico": true,
  "_icon_service_csp": "",
  "_icon_service_url": "",
  "_ip_header_enabled": true,
  "_smtp_img_src": "cid:",
  "admin_ratelimit_max_burst": 3,
  "admin_ratelimit_seconds": 300,
  "admin_session_lifetime": 20,
  "admin_token": "***",
  "allowed_iframe_ancestors": "",
  "attachments_folder": "data/attachments",
  "auth_request_purge_schedule": "30 * * * * *",
  "authenticator_disable_time_drift": false,
  "data_folder": "data",
  "database_conn_init": "",
  "database_max_conns": 10,
  "database_timeout": 30,
  "database_url": "***************",
  "db_connection_retries": 15,
  "disable_2fa_remember": false,
  "disable_admin_token": false,
  "disable_icon_download": false,
  "domain": "*****://******************",
  "domain_origin": "*****://******************",
  "domain_path": "",
  "domain_set": true,
  "duo_host": null,
  "duo_ikey": null,
  "duo_skey": null,
  "email_attempts_limit": 3,
  "email_change_allowed": true,
  "email_expiration_time": 600,
  "email_token_size": 6,
  "emergency_access_allowed": true,
  "emergency_notification_reminder_schedule": "0 3 * * * *",
  "emergency_request_timeout_schedule": "0 7 * * * *",
  "enable_db_wal": true,
  "event_cleanup_schedule": "0 10 0 * * *",
  "events_days_retain": null,
  "experimental_client_feature_flags": "fido2-vault-credentials",
  "extended_logging": true,
  "helo_name": null,
  "hibp_api_key": null,
  "icon_blacklist_non_global_ips": true,
  "icon_blacklist_regex": null,
  "icon_cache_folder": "data/icon_cache",
  "icon_cache_negttl": 259200,
  "icon_cache_ttl": 2592000,
  "icon_download_timeout": 10,
  "icon_redirect_code": 302,
  "icon_service": "internal",
  "incomplete_2fa_schedule": "30 * * * * *",
  "incomplete_2fa_time_limit": 3,
  "invitation_expiration_hours": 120,
  "invitation_org_name": "Vaultwarden",
  "invitations_allowed": true,
  "ip_header": "X-Real-IP",
  "job_poll_interval_ms": 30000,
  "log_file": null,
  "log_level": "Info",
  "log_timestamp_format": "%d-%m-%Y %H:%M:%S.%3f",
  "login_ratelimit_max_burst": 10,
  "login_ratelimit_seconds": 60,
  "org_attachment_limit": null,
  "org_creation_users": "",
  "org_events_enabled": false,
  "org_groups_enabled": false,
  "password_hints_allowed": true,
  "password_iterations": 600000,
  "push_enabled": false,
  "push_identity_uri": "https://identity.bitwarden.com",
  "push_installation_id": "***",
  "push_installation_key": "***",
  "push_relay_uri": "https://push.bitwarden.com",
  "reload_templates": false,
  "require_device_email": false,
  "rsa_key_filename": "data/rsa_key",
  "send_purge_schedule": "0 5 * * * *",
  "sendmail_command": null,
  "sends_allowed": true,
  "sends_folder": "data/sends",
  "show_password_hint": false,
  "signups_allowed": true,
  "signups_domains_whitelist": "",
  "signups_verify": false,
  "signups_verify_resend_limit": 6,
  "signups_verify_resend_time": 3600,
  "smtp_accept_invalid_certs": false,
  "smtp_accept_invalid_hostnames": false,
  "smtp_auth_mechanism": null,
  "smtp_debug": false,
  "smtp_embed_images": true,
  "smtp_explicit_tls": null,
  "smtp_from": "",
  "smtp_from_name": "Vaultwarden",
  "smtp_host": null,
  "smtp_password": null,
  "smtp_port": 587,
  "smtp_security": "starttls",
  "smtp_ssl": null,
  "smtp_timeout": 15,
  "smtp_username": null,
  "templates_folder": "data/templates",
  "tmp_folder": "data/tmp",
  "trash_auto_delete_days": null,
  "trash_purge_schedule": "0 5 0 * * *",
  "use_sendmail": false,
  "use_syslog": false,
  "user_attachment_limit": null,
  "user_send_limit": null,
  "web_vault_enabled": true,
  "web_vault_folder": "web-vault/",
  "websocket_address": "0.0.0.0",
  "websocket_enabled": false,
  "websocket_port": 3012,
  "yubico_client_id": null,
  "yubico_secret_key": null,
  "yubico_server": null
}
```
</details>

### Steps to reproduce
Press de sync button
![image](https://github.com/dani-garcia/vaultwarden/assets/46854222/78cae7fa-d200-49f9-a27f-75634440ee59)

I now detect than randomly sync, i think where the password must be sent on a field, the plugin automatically sync the database, 
### Expected behaviour
the plugin database must be synced 

### Actual behaviour
apper a sync error

### Troubleshooting data
![Screenshot_20240515_111436](https://github.com/dani-garcia/vaultwarden/assets/46854222/062330a1-5c5a-417b-975a-dd0030b45992)
![Screenshot_20240515_111432](https://github.com/dani-garcia/vaultwarden/assets/46854222/f39851b5-45c6-4d11-8854-f0b5268de798)
![Screenshot_20240515_102508](https://github.com/dani-garcia/vaultwarden/assets/46854222/69a86550-e997-46ea-be3d-61723ddffc67)

**Comment by Crosus97** [comment]
sorry i miss create this issue
now has the correct information

**Comment by Crosus97** [comment]
i dont know wy this is a discussion i think is an error

**Reply by BlackDex** [reply]
If this was a general issue, there would have been more reports to be fair.
For me syncing works just fine on FireFox, Android, Chromium, Edge, Bitwarden Desktop, iOS.

Please check the Vaultwarden and reverse proxy logs during a sync attempt and report what those are outputting.

**Reply by BlackDex** [reply]
Also, the reason Vaultwarden isn't seeing that a reverse proxy is used is that you probably do not have set a valid `IP_HEADER` in Vaultwarden which HAProxy might be sending.

**Reply by Crosus97** [reply]
I need to add this env variable and set the ip of de haproxy right?

**Reply by BlackDex** [reply]
No, it needs to be set to a header which HAProxy will send with the original IP address of the actual client which tries to access the server.
Else Vaultwarden will only see the IP of HAProxy, and never the one of the client.

**Comment by Crosus97** [comment]
i set the header, i think i solve the problem with chrome plugin do the forwarder-for on HAPROXY in pfsense, but i still having the android problem

i use this discussion, and now my notifications work on my android, but didnt syncronize the data to the phone
https://github.com/dani-garcia/vaultwarden/discussions/2035

but i dont know what to do with the android

https://gist.github.com/Crosus97/836c54b414878ca33a6cde6ca4e0794f

Discalimer:

The version and the captures of the problem are identical to the original posted

Discalimer2:
I test in my work connection, with 5G ,wifi home....

**Reply by BlackDex** [reply]
What are the logs reporting during the time you try to sync?
Without this information you are shooting in the dark.

**Comment by Crosus97** [comment]
i will update the gist

This log is from docker console, this is the order
https://gist.github.com/Crosus97/836c54b414878ca33a6cde6ca4e0794f
- Restart container
- Reinstall bitwarden on android
- Set selfhosted server
- Login with the account
- No entries...
- Go to setting to force the sync

Sync Fail Error


### Your environment (Generated via diagnostics page)
* Vaultwarden version: v1.30.5
* Web-vault version: v2024.1.2b
* OS/Arch: linux/x86_64
* Running within a container: true (Base: Debian)
* Environment settings overridden: true
* Uses a reverse proxy: true
* IP Header check: true (X-Forwarded-For)
* Internet access: true
* Internet access via a proxy: false
* DNS Check: true
* Browser/Server Time Check: true
* Server/NTP Time Check: true
* Domain Configuration Check: true
* HTTPS Check: true
* Database type: SQLite
* Database version: 3.44.0
* Clients used: 
* Reverse proxy and version: 
* Other relevant information: 

### Config (Generated via diagnostics page)
<details><summary>Show Running Config</summary>

**Environment settings which are overridden:** ADMIN_TOKEN


```json
{
  "_duo_akey": null,
  "_enable_duo": true,
  "_enable_email_2fa": false,
  "_enable_smtp": true,
  "_enable_yubico": true,
  "_icon_service_csp": "",
  "_icon_service_url": "",
  "_ip_header_enabled": true,
  "_smtp_img_src": "cid:",
  "admin_ratelimit_max_burst": 3,
  "admin_ratelimit_seconds": 300,
  "admin_session_lifetime": 20,
  "admin_token": "***",
  "allowed_iframe_ancestors": "",
  "attachments_folder": "data/attachments",
  "auth_request_purge_schedule": "30 * * * * *",
  "authenticator_disable_time_drift": false,
  "data_folder": "data",
  "database_conn_init": "",
  "database_max_conns": 10,
  "database_timeout": 30,
  "database_url": "***************",
  "db_connection_retries": 15,
  "disable_2fa_remember": false,
  "disable_admin_token": false,
  "disable_icon_download": false,
  "domain": "*****://******************",
  "domain_origin": "*****://******************",
  "domain_path": "",
  "domain_set": true,
  "duo_host": null,
  "duo_ikey": null,
  "duo_skey": null,
  "email_attempts_limit": 3,
  "email_change_allowed": true,
  "email_expiration_time": 600,
  "email_token_size": 6,
  "emergency_access_allowed": true,
  "emergency_notification_reminder_schedule": "0 3 * * * *",
  "emergency_request_timeout_schedule": "0 7 * * * *",
  "enable_db_wal": true,
  "event_cleanup_schedule": "0 10 0 * * *",
  "events_days_retain": null,
  "experimental_client_feature_flags": "fido2-vault-credentials",
  "extended_logging": true,
  "helo_name": null,
  "hibp_api_key": null,
  "icon_blacklist_non_global_ips": true,
  "icon_blacklist_regex": null,
  "icon_cache_folder": "data/icon_cache",
  "icon_cache_negttl": 259200,
  "icon_cache_ttl": 2592000,
  "icon_download_timeout": 10,
  "icon_redirect_code": 302,
  "icon_service": "internal",
  "incomplete_2fa_schedule": "30 * * * * *",
  "incomplete_2fa_time_limit": 3,
  "invitation_expiration_hours": 120,
  "invitation_org_name": "Vaultwarden",
  "invitations_allowed": true,
  "ip_header": "X-Forwarded-For",
  "job_poll_interval_ms": 30000,
  "log_file": null,
  "log_level": "debug",
  "log_timestamp_format": "%d-%m-%Y %H:%M:%S.%3f",
  "login_ratelimit_max_burst": 10,
  "login_ratelimit_seconds": 60,
  "org_attachment_limit": null,
  "org_creation_users": "",
  "org_events_enabled": false,
  "org_groups_enabled": false,
  "password_hints_allowed": true,
  "password_iterations": 600000,
  "push_enabled": false,
  "push_identity_uri": "https://identity.bitwarden.com",
  "push_installation_id": "***",
  "push_installation_key": "***",
  "push_relay_uri": "https://push.bitwarden.com",
  "reload_templates": false,
  "require_device_email": false,
  "rsa_key_filename": "data/rsa_key",
  "send_purge_schedule": "0 5 * * * *",
  "sendmail_command": null,
  "sends_allowed": true,
  "sends_folder": "data/sends",
  "show_password_hint": false,
  "signups_allowed": true,
  "signups_domains_whitelist": "",
  "signups_verify": false,
  "signups_verify_resend_limit": 6,
  "signups_verify_resend_time": 3600,
  "smtp_accept_invalid_certs": false,
  "smtp_accept_invalid_hostnames": false,
  "smtp_auth_mechanism": null,
  "smtp_debug": false,
  "smtp_embed_images": true,
  "smtp_explicit_tls": null,
  "smtp_from": "",
  "smtp_from_name": "Vaultwarden",
  "smtp_host": null,
  "smtp_password": null,
  "smtp_port": 587,
  "smtp_security": "starttls",
  "smtp_ssl": null,
  "smtp_timeout": 15,
  "smtp_username": null,
  "templates_folder": "data/templates",
  "tmp_folder": "data/tmp",
  "trash_auto_delete_days": null,
  "trash_purge_schedule": "0 5 0 * * *",
  "use_sendmail": false,
  "use_syslog": false,
  "user_attachment_limit": null,
  "user_send_limit": null,
  "web_vault_enabled": true,
  "web_vault_folder": "web-vault/",
  "websocket_address": "0.0.0.0",
  "websocket_enabled": false,
  "websocket_port": 3012,
  "yubico_client_id": null,
  "yubico_secret_key": null,
  "yubico_server": null
}
```
</details>


I use a PFSENSE and HAPROXY for my services, included vaultwarden

i use this guide tu set the header_ip setting:
https://vaultwarden.discourse.group/t/x-forwarded-for-issue/72/4

and this for the websocket
https://vaultwarden.discourse.group/t/haproxy-configuration-assistance/1027

**Reply by BlackDex** [reply]
That websocket config is outdated and will stop working the next upcoming release. 

Besides that, i see no errors in the log at all. So from that point of view, all seems to be working as it should. 

The next step would be to check the logs of your phone in some way. Maybe it is the encryption settings? If they are too high, some devices can't handle that.

**Reply by Crosus97** [reply]
its a samsung S21 ultra, the weird thing is that all was working 3 or 4 days ago, i did deploy the vaultwarden, configure the HAPROXY and start use it, i didn't do more config,because of this sync fail error, I started to configure the header_ip and everything else. that's what I find most strange, do you think it would solve something by reinstalling the container? could you guide me or send me some documentation to do it safely? it also worked because I migrated everything to vaultwarden, and I do not want to be left without the data. I have watchtower, I don't know if it might have been updated a few days ago or something.

**Reply by Crosus97** [reply]
is there any way to see the logs of the bitwarden application, by the way I also tried to use the beta version of the app but it didn't work either.

**Comment by Crosus97** [comment]
I FOUND IT

ITS A BUG

i read this discussion #3580 

and i found it 

i created a Secure Note to install a service with an openAIkey and i didnt imaging that can be the mistake

![image](https://github.com/dani-garcia/vaultwarden/assets/46854222/dc877de6-f644-44f9-94b6-07f34c0e6cd9)


and deleting the secure note,and delete it from the tras, the phone start sync again, magically

the step to reproduce is with my config create a secure note and the bug appear 
![image](https://github.com/dani-garcia/vaultwarden/assets/46854222/d75909cb-4ed9-495c-a52d-8aee1975e94d)
